### PR TITLE
Redesign timeline sample pieces

### PR DESCRIPTION
**Pull request by OliBomby**
Changed `SamplePointPiece` to show more hitsounding information on the timeline in a concise manner. You can now also see which additions are enabled and if there is a different additions bank name.

I'm not certain about my choice of colors, but I at least know that the circles indicating which additions are enabled should have different colors to make it more readable.

Also the timeline is now significantly higher so it might be good to add a toggle for hitsound info visibility on the timeline in the future. I know only about 50% of mappers apply hitsound while mapping.

It's also possible to later add the ability to edit the volume sliders on these right from the timeline, so a mapper could make volume gradients in a single gesture.

Depends on 
- [x] https://github.com/ppy/osu/pull/23443

image

### CODE REVIEW

**Comment by bdach** [comment]
Continuing [discussion from discord](https://discord.com/channels/188630481301012481/188630652340404224/1105257216975577129) an extremely crude mockup of my counter-proposal of how this could look follows:



- Does not overload colours used elsewhere in editor already
- Icons correspond to the ones already used on the left sidebar
- Bright pink = sound on, dark pink = sound off
- Letters next to each sound are the first letter of the bank name. (Yes I know stable doesn't support this, it only supports a bank for the main sound and a bank for everything else, but I think that is weird and bad)

**Comment by OliBomby** [comment]
> Continuing [discussion from discord](https://discord.com/channels/188630481301012481/188630652340404224/1105257216975577129) an extremely crude mockup of my counter-proposal of how this could look follows:
> 
> 
> 
> * Does not overload colours used elsewhere in editor already
> * Icons correspond to the ones already used on the left sidebar
> * Bright pink = sound on, dark pink = sound off
> * Letters next to each sound are the first letter of the bank name. (Yes I know stable doesn't support this, it only supports a bank for the main sound and a bank for everything else, but I think that is weird and bad. But what you _can_ do is display it in the way we eventually want to already, which is bank per sound)

I gotta admit your proposal does look much easier to initially understand than my design. The icons and the associated bank name make it clear which bank goes where and what addition hitsound is used. However there are two downsides:

- Your proposal requires significantly more width, which makes it more likely for them to overlap with neighbours and becoming unreadable. Also extra width makes the problem worse of the first node sample and the sliderslide sample having their sample timeline piece on the same tick.
- The addition icons are always visible, which might make the timeline seem even more cluttered.

So do we want a more concise design which is as narrow as possible or a more verbose design which is less ambiguous?
I personally prefer a concise design because you can still click the sample piece to see all the details in the popup.

**Comment by bdach** [comment]
I dunno. On the screenshots above everything is so tiny that it just makes me think that we aren't operating at the correct density of information. If it can't be added using a size large enough to be legible then it likely just doesn't belong on the timeline at all, or at least not in the general case (it may be fine if there is a special view for hitsounding or something, which enlarges the timeline enough wherein the information can be legible).

**Comment by OliBomby** [comment]
With a fresh perspective after 1 year. I think the last screenshot looks best. It's legible and conveys the necessary information without adding too much clutter.

I'd make the volume bar a bit shorter so it matches the samples on the right.

If we keep the compose timeline for basic hitsounds without custom sample banks, like I suggested in https://github.com/ppy/osu/pull/23443#issuecomment-2173078067, then its sufficient to only visualize the normal, soft, drum sample banks and the standard additions.

**Comment by bdach** [comment]
I dunno about any of these to be honest.

Every single actual screenshot contains either one of:
- elements that are too small to read comfortably
- elements that require significant mental processing to decipher (by which I mean having to decode coloured dots to banks)
- generally unsure of having the timeline take up _more_ of screen space when already people are saying that the playfield is not big enough and elements like the bottom timeline are too large for what they provide
- also not sure the bars for volume are legible if they don't show the actual value, it seems difficult to get a correct reference frame for how loud hitsounds are in relation to each other

Basically I'm not happy about any of these proposals to say in earnest that I'd be fine with merging either. That said I'm also not sure I have anything better to offer at this time.
